### ISSUE TITLE
`manual_let_else` not suggesting a fix for match statement

### ISSUE THREAD

**Issue by nyurik** [rust-lang/rust-clippy#14598]
### Summary

An expected suggistion was not made

### Lint Name

manual_let_else

### Reproducer

I tried this code:

```rust
fn main(){
    assert!(bar().is_ok());
}

fn bar() -> Result<bool, &'static str> {
    let value = match foo() {
        Err(_) => return Err("abc"),
        Ok(value) => value,
    };

    Ok(value == 42)
}

#[allow(clippy::unnecessary_wraps)]
fn foo() -> Result<u32, &'static str>  {
    Ok(42)
}

```

I expected this suggestion for the `let value = ...` statement:

```rust
    let Ok(value) = foo() else {
        return Err("abc");
    };
```

Instead, there were no suggestions.

### Version

```text
rustc 1.86.0 (05f9846f8 2025-03-31)
binary: rustc
commit-hash: 05f9846f893b09a1be1fc8560e33fc3c815cfecb
commit-date: 2025-03-31
host: x86_64-unknown-linux-gnu
release: 1.86.0
LLVM version: 19.1.7
```
